test/e2e/alertmanager_test.go

```diff
@@ -161,3 +170,220 @@ func TestAlertmanagerTrustedCA(t *testing.T) {
+	const testNs = "test-kube-rbac-proxy"
+
+	// The tenancy port (9092) is only exposed in-cluster so we need to use
+	// port forwarding to access kube-rbac-proxy.
+	host, cleanUp, err := f.ForwardPort("alertmanager-main", 9092)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanUp()
+
+	t.Logf("Creating namespace %q", testNs)
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testNs,
+		},
+	}
+	ns, err = f.KubeClient.CoreV1().Namespaces().Create(ns)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		err := f.KubeClient.CoreV1().Namespaces().Delete(testNs, &metav1.DeleteOptions{})
+		t.Logf("Deleting namespace %s: %v", testNs, err)
+	}()
+
+	// Creating service accounts with different role bindings.
+	clients := make(map[string]silenceClient)
+	for sa, cr := range map[string]string{
+		"editor":    "monitoring-rules-edit",
+		"viewer":    "monitoring-rules-view",
+		"anonymous": "",
+	} {
+		t.Logf("Creating service account %q", sa)
+		_, err = f.CreateServiceAccount(testNs, sa)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if cr != "" {
+			t.Logf("Creating role binding %q -> %q", sa, cr)
+			_, err = f.CreateRoleBindingFromClusterRole(testNs, sa, cr)
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+
+		err = framework.Poll(5*time.Second, 5*time.Minute, func() error {
+			secrets, err := f.KubeClient.CoreV1().Secrets(testNs).List(metav1.ListOptions{})
+			if err != nil {
+				return err
+			}
+			for _, secret := range secrets.Items {
+				_, dockerToken := secret.Annotations["openshift.io/create-dockercfg-secrets"]
+				token := strings.Contains(secret.Name, fmt.Sprintf("%s-token-", sa))
+
+				// we have to skip the token secret that contains the openshift.io/create-dockercfg-secrets annotation
+				// as this is the token to talk to the internal registry.
+				if !dockerToken && token {
+					clients[sa] = silenceClient{
+						t:         t,
+						host:      host,
+						token:     string(secret.Data["token"]),
+						namespace: testNs,
+					}
+					return nil
+				}
+			}
+			return errors.Errorf("couldn't find token secret for service account %q", sa)
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// Create a silence.
+	now := time.Now()
+	sil := []byte(fmt.Sprintf(
+		`{"matchers":[{"name":"foo","value":"bar","isRegex":false}],"startsAt":"%s","endsAt":"%s","createdBy":"somebody","comment":"some comment"}`,
+		now.Format(time.RFC3339),
+		now.Add(time.Hour).Format(time.RFC3339),
+	))
+
+	for _, sa := range []string{"viewer", "anonymous"} {
+		t.Logf("Creating silence as %q (denied)", sa)
+		_ = clients[sa].do("POST", "/api/v2/silences", sil, http.StatusForbidden)
+	}
+
+	t.Log("Creating silence as 'editor' (allowed)")
+	b := clients["editor"].do("POST", "/api/v2/silences", sil, http.StatusOK)
+
+	// Save silence ID for deletion.
+	parsed, err := gabs.ParseJSON(b)
+	if err != nil {
+		t.Fatalf("%v (data: %q)", err, string(b))
+	}
+	silID, ok := parsed.Path("silenceID").Data().(string)
```

Comment:
let's additionally validate that the namespace label exists no?

Reply:
done, the original silence is created with a `namespace="openshift-monitoring"` matcher and the test verifies that it is modified to  `namespace="test-kube-rbac-proxy"`.